### PR TITLE
Patching binutils 2.31.1 and 2.32 to fix compatibility with RHEL8

### LINKED ISSUE #9182: Linker problems with Bazel-0.25.2-GCCcore-8.2.0 on RHEL8
Building Bazel causes linker problems, similar to what has been seen when using ld.gold in other situations. Building with the SYSTEM toolchain works. The relevant part of the logs are:

```
ERROR: /tmp/eb-v7mb3x3f/bazel_bmkm1c5Q/out/external/bazel_tools/third_party/ijar/BUILD:74:1: Linking of rule '@bazel_tools//third_party/ijar:zipper' failed (Exit 1)
: gcc failed: error executing command
  (cd /tmp/eb-v7mb3x3f/bazel_bmkm1c5Q/out/execroot/io_bazel && \
  exec env - \
    LD_LIBRARY_PATH=/opt/uio/modules/rhel8/easybuild/software/Java/1.8.0_221/lib:/opt/uio/modules/rhel8/easybuild/software/binutils/2.31.1-GCCcore-8.2.0/lib:/opt/uio/modules/rhel
8/easybuild/software/zlib/1.2.11-GCCcore-8.2.0/lib:/opt/uio/modules/rhel8/easybuild/software/GCCcore/8.2.0/lib/gcc/x86_64-pc-linux-gnu/8.2.0:/opt/uio/modules/rhel8/easybuild/soft
ware/GCCcore/8.2.0/lib64:/opt/uio/modules/rhel8/easybuild/software/GCCcore/8.2.0/lib \
    PATH=/opt/uio/modules/rhel8/easybuild/software/Java/1.8.0_221:/opt/uio/modules/rhel8/easybuild/software/Java/1.8.0_221/bin:/opt/uio/modules/rhel8/easybuild/software/binutils/
2.31.1-GCCcore-8.2.0/bin:/opt/uio/modules/rhel8/easybuild/software/GCCcore/8.2.0/bin:/opt/uio/modules/rhel8/easybuild/packages/EasyBuild/4.0.1/bin:/site/bin:/site/common/hosts:/u
sr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin \
    PWD=/proc/self/cwd \
  /opt/uio/modules/rhel8/easybuild/software/GCCcore/8.2.0/bin/gcc -o bazel-out/host/bin/external/bazel_tools/third_party/ijar/zipper -Wl,-S '-fuse-ld=gold' -Wl,-no-as-needed -Wl,
-z,relro,-z,now -B/opt/uio/modules/rhel8/easybuild/software/GCCcore/8.2.0/bin -pass-exit-codes -Wl,--gc-sections -Wl,@bazel-out/host/bin/external/bazel_tools/third_party/ijar/zip
per-2.params)
Execution platform: @bazel_tools//platforms:host_platform
[868 / 903] 5 actions running
    //third_party/protobuf/3.6.1:protobuf_lite; 0s local
    //third_party/grpc:grpc++_base_unsecure; 0s local
    Compiling .../3.6.1/src/google/protobuf/io/zero_copy_stream.cc; 0s local
    Compiling .../grpc/src/core/lib/compression/message_compress.cc; 0s local
    Linking third_party/ijar/ijar [for host]; 0s local
^M^M^M^M^M^M/lib/../lib64/crti.o(.gnu.build.attributes+0x14): error: relocation refers to local symbol "" [1], which is defined in a discarded section
/lib/../lib64/crti.o(.gnu.build.attributes+0x1c): error: relocation refers to local symbol "" [1], which is defined in a discarded section
/lib/../lib64/crtn.o(.gnu.build.attributes+0x14): error: relocation refers to local symbol "" [1], which is defined in a discarded section
/lib/../lib64/crtn.o(.gnu.build.attributes+0x1c): error: relocation refers to local symbol "" [1], which is defined in a discarded section
/lib/../lib64/crt1.o(.gnu.build.attributes.hot+0x18): error: relocation refers to local symbol "" [2], which is defined in a discarded section
/lib/../lib64/crt1.o(.gnu.build.attributes.hot+0x20): error: relocation refers to local symbol "" [2], which is defined in a discarded section
/lib/../lib64/crt1.o(.gnu.build.attributes.unlikely+0x18): error: relocation refers to local symbol "" [3], which is defined in a discarded section
/lib/../lib64/crt1.o(.gnu.build.attributes.unlikely+0x20): error: relocation refers to local symbol "" [3], which is defined in a discarded section
/lib/../lib64/crt1.o(.gnu.build.attributes.hot+0x18): error: relocation refers to local symbol "" [4], which is defined in a discarded section
/lib/../lib64/crt1.o(.gnu.build.attributes.hot+0x20): error: relocation refers to local symbol "" [4], which is defined in a discarded section
/lib/../lib64/crt1.o(.gnu.build.attributes.unlikely+0x18): error: relocation refers to local symbol "" [5], which is defined in a discarded section
/lib/../lib64/crt1.o(.gnu.build.attributes.unlikely+0x20): error: relocation refers to local symbol "" [5], which is defined in a discarded section
/usr/lib64/libc_nonshared.a(elf-init.oS)(.gnu.build.attributes.hot+0x18): error: relocation refers to local symbol "" [4], which is defined in a discarded section
/usr/lib64/libc_nonshared.a(elf-init.oS)(.gnu.build.attributes.hot+0x20): error: relocation refers to local symbol "" [4], which is defined in a discarded section
/usr/lib64/libc_nonshared.a(elf-init.oS)(.gnu.build.attributes.unlikely+0x18): error: relocation refers to local symbol "" [7], which is defined in a discarded section
/usr/lib64/libc_nonshared.a(elf-init.oS)(.gnu.build.attributes.unlikely+0x20): error: relocation refers to local symbol "" [7], which is defined in a discarded section
collect2: error: ld returned 1 exit status
```

### PR DESCRIPTION
**Pull request by terjekv**
(created using `eb --new-pr`)

fixes #9182

### CODE REVIEW

File: easybuild/easyconfigs/b/binutils/binutils-2.31.1-gold-ignore-discarded-note-relocts.patch [hunk 1]
```diff
@@ -0,0 +1,11 @@
+diff -ru binutils-2.31.1.orig/gold/target-reloc.h binutils-2.31.1/gold/target-reloc.h
```
Comment (boegel) [inline easybuild/easyconfigs/b/binutils/binutils-2.31.1-gold-ignore-discarded-note-relocts.patch]: @terjekv Please add a short patch description on top of this file (anything above the `---` line is ignored by `patch`), as well as a reference to where this patch was obtained from.

Keep in mind that this implies updating the SHA256 checksums as well

Comment (terjekv) [reply easybuild/easyconfigs/b/binutils/binutils-2.31.1-gold-ignore-discarded-note-relocts.patch]: Comments added. Tested by rebuilding all affected versions of binutils, and then rebuilding LLVM and Bazel that caused the errors beforehand.

**Comment by boegel** [comment]
@terjekv You need to update the SHA256 checksums as well after adding the comments to the patch file...

**Comment by terjekv** [comment]
> @terjekv You need to update the SHA256 checksums as well after adding the comments to the patch file...

Oh I did update the EBs, I merely forgot to upload them... The one time I missed git status.  ;-)

**Comment by boegel** [comment]
@terjekv (still) works fine on CentOS 7, when building everything from scratch (see upcoming test report)

I'll test a couple of things on top of this as well (incl. Bazel)

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 7 out of 7 (7 easyconfigs in this PR)
node3112.skitty.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/94ff1d80c3937588a3a02f716d4133fe for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 7 out of 7 (7 easyconfigs in this PR)
node2595.golett.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/ddb07089e8c50216cd0b51f384732cdd for a full test report.

**Comment by terjekv** [comment]
Test report by @terjekv
**SUCCESS**
Build succeeded for 0 out of 0 (7 easyconfigs in this PR)
ninhursaga.uio.no - Linux RHEL 8.1, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz, Python 2.7.16
See https://gist.github.com/c8396beccb838f83a30def14bbe38e94 for a full test report.

**Comment by terjekv** [comment]
Test report by @terjekv
**SUCCESS**
Build succeeded for 7 out of 7 (7 easyconfigs in this PR)
ninhursaga.uio.no - Linux RHEL 8.1, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz, Python 2.7.16
See https://gist.github.com/7de1121ca15cf2ceeea263aaf24d662c for a full test report.

**Comment by terjekv** [comment]
Hm, doing a complete test and rebuilding the entire environment, as well as Bazel and LLVM, then uploading the results. Apologies for the spam.

**Comment by boegel** [comment]
I tested building `TensorFlow-2.0.0-foss-2019a-Python-3.7.2.eb` + all required dependencies (incl. `Bazel-0.26.1-GCCcore-8.2.0.eb` from scratch on CentOS 7.7, (still) works like a charm.

**Comment by terjekv** [comment]
Test report by @terjekv
**SUCCESS**
Build succeeded for 66 out of 66 (9 easyconfigs in this PR)
ninhursaga.uio.no - Linux RHEL 8.1, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz, Python 2.7.16
See https://gist.github.com/87ff1f1b93bb2045089fb164c1ea9bd5 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @terjekv!

**Pull request merged**